APMrover2/sailboat.h

```diff
@@ -51,14 +52,30 @@ class Sailboat
     void  clear_tack();
 
     // returns true if boat is currently tacking
-    bool  tacking() const;
+    bool  tacking();
 
     // returns true if sailboat should take a indirect navigation route to go upwind
-    bool  use_indirect_route(float desired_heading_cd) const;
+    bool  use_indirect_route(float desired_heading_cd);
 
     // calculate the heading to sail on if we cant go upwind
     float calc_heading(float desired_heading_cd);
 
+    // return sailboat loiter radius
+    float get_loiter_radius() const {return loit_radius;}
+
+    // update throttle for manual throttle modes
+    float update_manual_throttle(float desired_throttle);
+
+    // check aux throttle at arming
+    bool aux_throttle_pre_arm_check();
+
+    enum sailboat_throttle {
+        NEVER        = 0,
+        ASSIST       = 1,
+        FORCE_MOTOR  = 2
+    };
+    uint8_t throttle_state; // throttle state used with throttle enum
```

Comment:
if throttle state should always be one of the sailboat_throttle values then might as well make it's type "enum sailboat_throttle".  There's probably a better name for this enum though.  Often we use camel case for enums (although not always).  If it's going to use the underscore format then it should probably end with "_t" I think.

I would normally hide the set behind an accessor but that's probably not necessary and it eats some flash so it's probably ok.

Reply:
fixed